include/nbl/video/utilities/IGPUVirtualTexture.h

```diff
@@ -38,24 +38,25 @@ class IGPUVirtualTexture final : public asset::IVirtualTexture<IGPUImageView, IG
         uint32_t& waitSemaphoreCount, IGPUSemaphore* const* &semaphoresToWaitBeforeExecution, const asset::E_PIPELINE_STAGE_FLAGS* &stagesToWaitForPerSemaphore
     )
     {
+        core::smart_refctd_ptr<IUtilities> utilities = core::make_smart_refctd_ptr<IUtilities>(core::smart_refctd_ptr<ILogicalDevice>(logicalDevice));
 
         core::smart_refctd_ptr<IGPUImage> gpuImage;
         {
             auto cpuImageParams = _cpuimg->getCreationParameters();
             cpuImageParams.initialLayout = asset::EIL_TRANSFER_DST_OPTIMAL;
-        
+
             // TODO: Look at issue #167 on Nabla repo, at some point
-            auto gpuTexelBuffer = logicalDevice->createDeviceLocalGPUBufferOnDedMem(_cpuimg->getBuffer()->getSize());
+            auto gpuTexelBuffer = logicalDevice->createDeviceLocalGPUBufferOnDedMem(IGPUBuffer::SCreationParams(), _cpuimg->getBuffer()->getSize());
             uint32_t signalSemaphoreCount=0u;
             IGPUSemaphore* const* semaphoresToSignal = nullptr;
-            logicalDevice->updateBufferRangeViaStagingBuffer(
+            utilities->updateBufferRangeViaStagingBuffer(
                 cmdbuf,fence,queue,asset::SBufferRange<IGPUBuffer>{0u,gpuTexelBuffer->getSize(),gpuTexelBuffer},_cpuimg->getBuffer()->getPointer(),
                 waitSemaphoreCount,semaphoresToWaitBeforeExecution,stagesToWaitForPerSemaphore
             );
 
             auto regions = _cpuimg->getRegions();
             assert(regions.size());
-            gpuImage = logicalDevice->createFilledDeviceLocalGPUImageOnDedMem(cmdbuf,std::move(cpuImageParams),gpuTexelBuffer.get(),regions.size(),regions.begin());
+            gpuImage = utilities->createFilledDeviceLocalGPUImageOnDedMem(cmdbuf,std::move(cpuImageParams),gpuTexelBuffer.get(),regions.size(),regions.begin());
```

Comment:
you actually need to submit the commandbuffer while signalling the fence later on, are you doing that?